ui/app/adapters/keymgmt/key.js

```diff
@@ -74,32 +74,60 @@ export default class KeymgmtKeyAdapter extends ApplicationAdapter {
     return this._updateKey(backend, name, data);
   }
 
+  distribute(backend, kms, key, data) {
+    return this.ajax(`${this.buildURL()}/${backend}/kms/${encodePath(kms)}/key/${encodePath(key)}`, 'PUT', {
```

Comment:
The `encodePath` piece is just making the key URL safe. There isn't a lookup because the path wouldn't necessarily have all the information; the distribution form allows providing both the provider and the key name, so calling this from anywhere would work assuming you provide values that map to keymgmt items that exist within vault.

Reply:
Cool thanks for the information